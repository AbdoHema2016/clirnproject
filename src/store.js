import {createStore, applyMiddleware} from 'redux';
import createSagaMiddleware from 'redux-saga';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import {rootSaga} from './sagas';
const sagaMiddleware = createSagaMiddleware();
const isDevelopmentEnv =
  process && process.env && process.env.NODE_ENV === 'development';
const isProductionEnv =
  process && process.env && process.env.NODE_ENV === 'production';
const logStoreInfo = false;
const middlewares = [sagaMiddleware];
function actionLogger({getState}) {
  return (next) => (action) => {
    let allowed = [
      'jumio',
      'notificationsReducer',
      'otp',
      'companies',
      'signIn',
      'updateContactDetailsReducer',
      'userProfile',
      'vaccinations',
    ];
    let state = getState();
    let loggedStates = Object.keys(state)
      .filter((key) => allowed.includes(key))
      .reduce((obj, key) => {
        return {
          ...obj,
          [key]: state[key],
        };
      }, {});
    console.info({actionType: action.type, loggedStates});
    const returnValue = next(action);
    return returnValue;
  };
}
if (isDevelopmentEnv && logStoreInfo) {
  middlewares.push(createLogger());
}
if (isProductionEnv) {
  middlewares.push(actionLogger);
}

const store = createStore(rootReducer, applyMiddleware(...middlewares));
sagaMiddleware.run(rootSaga);
export default store;
