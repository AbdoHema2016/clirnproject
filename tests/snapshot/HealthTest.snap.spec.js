import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import store from '../../src/store';
import HealthTest from '../../src/containers/health/healthTest/HealthTest';

jest.mock('@react-navigation/native', () => {});
jest.mock('react-native-gesture-handler', () => {});

jest.mock('@react-navigation/native', () => ({
  withNavigationFocus: (component) => component,
}));

jest.useFakeTimers();

describe('HealthTest screen', () => {
  const props = {
    navigation: {
      setOptions: jest.fn(),
    },
    addContactetails: jest.fn(),
    resendOtp: jest.fn(),
    emptyOtp: jest.fn(),
    resetOtp: jest.fn(),
    sendOtp: jest.fn(),
  };
  it('Snapshot HealthTest', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <HealthTest {...props} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
