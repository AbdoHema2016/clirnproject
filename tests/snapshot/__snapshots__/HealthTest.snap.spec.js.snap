// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HealthTest screen Snapshot HealthTest 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F7F7F7",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "height": 1063.254,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "height": "100%",
          "width": 724,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#8D8C91",
              "fontFamily": "SofiaPro-Medium",
              "fontSize": 16,
              "marginTop": 160.07999999999998,
            },
          ]
        }
      >
        7/7
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#2B342C",
              "fontFamily": "SofiaPro-Medium",
              "fontSize": 24,
              "fontWeight": "600",
            },
          ]
        }
      >
        [missing "en.healthTest.didYouHaveHealthTest" translation]
      </Text>
      <RCTScrollView
        data={
          Array [
            Object {
              "testData": Object {
                "checkMissingFieldsAtTheEnd": false,
                "document": Object {
                  "name": "",
                  "type": "",
                  "uri": "",
                },
                "howWasThistestPerformed": Object {
                  "empty": false,
                  "status": "",
                  "value": "A swab of my nose or throat",
                },
                "id": 1,
                "missingFields": true,
                "testCentre": Object {
                  "empty": true,
                  "status": "",
                  "value": "",
                },
                "testDate": Object {
                  "empty": true,
                  "status": "",
                  "value": "",
                },
                "testResult": Object {
                  "empty": false,
                  "status": "",
                  "value": "",
                },
                "testType": Object {
                  "empty": false,
                  "status": "",
                  "value": "",
                },
              },
            },
          ]
        }
        disableVirtualization={false}
        extraData={
          Object {
            "access_token": "",
            "addContactetails": [MockFunction],
            "addHealthDataFields": [Function],
            "checkMissingFields": [Function],
            "checkMissingFieldsInTestObject": [Function],
            "currentSelectedIndex": 0,
            "currentType": "",
            "deleteTestDetailRow": [Function],
            "editHealthData": [Function],
            "emptyOtp": [MockFunction],
            "id": "",
            "loading": false,
            "missingFields": true,
            "navigation": Object {
              "setOptions": [MockFunction] {
                "calls": Array [
                  Array [
                    Object {
                      "headerLeft": [Function],
                      "headerTransparent": true,
                      "title": "",
                    },
                  ],
                ],
                "results": Array [
                  Object {
                    "type": "return",
                    "value": undefined,
                  },
                ],
              },
            },
            "profileVisitedBeforeAction": [Function],
            "resendOtp": [MockFunction],
            "resetOtp": [MockFunction],
            "results": Array [
              Object {
                "options": Array [
                  "IgG only positive",
                  "IgG and IgM positive",
                  "IgM positive",
                  "IgG and IgM negative",
                ],
                "type": "Antibody test",
              },
              Object {
                "options": Array [
                  "Positive",
                  "Negative",
                ],
                "type": "Antigen test",
              },
              Object {
                "options": Array [
                  "Positive",
                  "Negative",
                ],
                "type": "RT PCR Test",
              },
            ],
            "sendOtp": [MockFunction],
            "showActionSheetWithOptions": [Function],
            "storeToken": [Function],
            "submitHealthTestData": [Function],
            "testDetails": Array [
              Object {
                "testData": Object {
                  "checkMissingFieldsAtTheEnd": false,
                  "document": Object {
                    "name": "",
                    "type": "",
                    "uri": "",
                  },
                  "howWasThistestPerformed": Object {
                    "empty": false,
                    "status": "",
                    "value": "A swab of my nose or throat",
                  },
                  "id": 1,
                  "missingFields": true,
                  "testCentre": Object {
                    "empty": true,
                    "status": "",
                    "value": "",
                  },
                  "testDate": Object {
                    "empty": true,
                    "status": "",
                    "value": "",
                  },
                  "testResult": Object {
                    "empty": false,
                    "status": "",
                    "value": "",
                  },
                  "testType": Object {
                    "empty": false,
                    "status": "",
                    "value": "",
                  },
                },
              },
            ],
            "types": Array [
              "Antibody test",
              "Antigen test",
              "RT PCR Test",
            ],
            "updateMissingFieldStatus": [Function],
            "updateTypeAndIndex": [Function],
            "uploadImageS3": [Function],
            "visitingProfileViaAction": [Function],
          }
        }
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={true}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Array [
                  Object {
                    "height": "auto",
                    "marginTop": 16,
                    "paddingBottom": 16,
                    "width": 724,
                  },
                  null,
                ]
              }
            >
              <View
                style={
                  Object {
                    "height": "auto",
                    "width": "auto",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#2B342C",
                        "fontFamily": "SofiaProRegular",
                        "fontSize": 12,
                        "height": 16,
                        "marginTop": 16,
                      },
                    ]
                  }
                >
                  Test centre*
                </Text>
                <View
                  style={
                    Array [
                      Object {
                        "height": "auto",
                        "width": 724,
                      },
                      null,
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#FFFFFF",
                          "borderRadius": 4,
                          "height": 48,
                          "justifyContent": "center",
                          "marginTop": 4,
                          "width": "100%",
                        },
                        undefined,
                        null,
                      ]
                    }
                  >
                    <TextInput
                      accessibilityLabel="Centre"
                      accessible={true}
                      allowFontScaling={true}
                      blurOnSubmit={true}
                      numberOfLines={1}
                      onChangeText={[Function]}
                      placeholder="Enter the name"
                      placeholderTextColor="#8D8C91"
                      rejectResponderTermination={true}
                      style={
                        Array [
                          Object {
                            "color": "#2B342C",
                            "fontSize": 18,
                            "fontWeight": "500",
                            "lineHeight": 20,
                            "paddingLeft": 19,
                            "paddingRight": 19,
                            "width": "100%",
                          },
                          Object {},
                        ]
                      }
                      testID="Centre"
                      underlineColorAndroid="transparent"
                      value=""
                    />
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "height": "auto",
                    "width": "auto",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#2B342C",
                        "fontFamily": "SofiaProRegular",
                        "fontSize": 12,
                        "height": 16,
                        "marginTop": 16,
                      },
                    ]
                  }
                >
                  Test type*
                </Text>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#FFFFFF",
                        "borderRadius": 4,
                        "flexDirection": "row",
                        "height": 48,
                        "marginTop": 4,
                      },
                      null,
                      null,
                    ]
                  }
                >
                  <View>
                    <View>
                      Antibody test
                    </View>
                    <View>
                      Antigen test
                    </View>
                    <View>
                      RT PCR Test
                    </View>
                  </View>
                  <Image
                    source={
                      Object {
                        "ReactComponent": [Function],
                      }
                    }
                    style={
                      Object {
                        "height": 9,
                        "position": "absolute",
                        "right": 18,
                        "width": 18,
                      }
                    }
                  />
                </View>
              </View>
              <View
                style={
                  Object {
                    "height": "auto",
                    "width": "auto",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#2B342C",
                        "fontFamily": "SofiaProRegular",
                        "fontSize": 12,
                        "height": 16,
                        "marginTop": 16,
                      },
                    ]
                  }
                >
                  Test date*
                </Text>
                <View>
                  <View
                    style={
                      Array [
                        Object {
                          "backgroundColor": "#FFFFFF",
                          "borderRadius": 4,
                          "flexDirection": "row",
                          "height": 48,
                        },
                        null,
                        null,
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "borderRadius": 4,
                            "height": 52,
                            "justifyContent": "center",
                            "shadowColor": "#DDDDDD",
                            "shadowOffset": Object {
                              "height": 8,
                              "width": 0,
                            },
                            "shadowOpacity": 0.75,
                            "shadowRadius": 4,
                          },
                          Object {
                            "height": "100%",
                            "shadowColor": "rgba(0,0,0,0)",
                            "width": "100%",
                          },
                        ]
                      }
                    >
                      <View
                        accessibilityLabel="TestDate"
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": undefined,
                            "borderRadius": 4,
                            "height": "100%",
                            "justifyContent": "center",
                            "opacity": 1,
                            "width": 724,
                          }
                        }
                        testID="TestDate"
                      >
                        <Text
                          numberOfLines={1}
                          style={
                            Array [
                              Object {
                                "fontFamily": "SofiaProRegular",
                                "fontSize": 17,
                                "textAlign": "center",
                              },
                              Object {
                                "color": "#8D8C91",
                                "left": 16,
                                "position": "absolute",
                              },
                            ]
                          }
                        >
                          XX/XX/XX
                        </Text>
                        <View
                          style={
                            Array [
                              Object {
                                "overflow": "hidden",
                              },
                              Array [
                                Object {
                                  "height": 18,
                                  "position": "absolute",
                                  "right": 18,
                                  "width": 18,
                                },
                                undefined,
                              ],
                            ]
                          }
                        >
                          <FastImageView
                            resizeMode="contain"
                            source={
                              Object {
                                "cache": "immutable",
                                "uri": undefined,
                              }
                            }
                            style={
                              Object {
                                "bottom": 0,
                                "left": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "height": "auto",
                    "width": "auto",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#2B342C",
                        "fontFamily": "SofiaProRegular",
                        "fontSize": 12,
                        "height": 16,
                        "marginTop": 16,
                      },
                    ]
                  }
                >
                  Test results*
                </Text>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#FFFFFF",
                        "borderRadius": 4,
                        "flexDirection": "row",
                        "height": 48,
                        "marginTop": 4,
                      },
                      null,
                      null,
                    ]
                  }
                >
                  <View />
                  <Image
                    source={
                      Object {
                        "ReactComponent": [Function],
                      }
                    }
                    style={
                      Object {
                        "height": 9,
                        "position": "absolute",
                        "right": 18,
                        "width": 18,
                      }
                    }
                  />
                </View>
              </View>
              <View
                style={
                  Object {
                    "height": "auto",
                    "width": "auto",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#2B342C",
                        "fontFamily": "SofiaProRegular",
                        "fontSize": 12,
                        "height": 16,
                        "marginTop": 16,
                      },
                    ]
                  }
                >
                  How was this test performed*
                </Text>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#FFFFFF",
                        "borderRadius": 4,
                        "flexDirection": "row",
                        "height": 48,
                        "marginTop": 4,
                      },
                      null,
                      null,
                    ]
                  }
                >
                  <View>
                    <View>
                      [
                    </View>
                    <View>
                      m
                    </View>
                    <View>
                      i
                    </View>
                    <View>
                      s
                    </View>
                    <View>
                      s
                    </View>
                    <View>
                      i
                    </View>
                    <View>
                      n
                    </View>
                    <View>
                      g
                    </View>
                    <View>
                       
                    </View>
                    <View>
                      "
                    </View>
                    <View>
                      e
                    </View>
                    <View>
                      n
                    </View>
                    <View>
                      .
                    </View>
                    <View>
                      t
                    </View>
                    <View>
                      e
                    </View>
                    <View>
                      s
                    </View>
                    <View>
                      t
                    </View>
                    <View>
                      M
                    </View>
                    <View>
                      e
                    </View>
                    <View>
                      t
                    </View>
                    <View>
                      h
                    </View>
                    <View>
                      o
                    </View>
                    <View>
                      d
                    </View>
                    <View>
                      s
                    </View>
                    <View>
                      "
                    </View>
                    <View>
                       
                    </View>
                    <View>
                      t
                    </View>
                    <View>
                      r
                    </View>
                    <View>
                      a
                    </View>
                    <View>
                      n
                    </View>
                    <View>
                      s
                    </View>
                    <View>
                      l
                    </View>
                    <View>
                      a
                    </View>
                    <View>
                      t
                    </View>
                    <View>
                      i
                    </View>
                    <View>
                      o
                    </View>
                    <View>
                      n
                    </View>
                    <View>
                      ]
                    </View>
                    <View>
                      A swab of my nose or throat
                    </View>
                  </View>
                  <Image
                    source={
                      Object {
                        "ReactComponent": [Function],
                      }
                    }
                    style={
                      Object {
                        "height": 9,
                        "position": "absolute",
                        "right": 18,
                        "width": 18,
                      }
                    }
                  />
                </View>
              </View>
              <View
                style={
                  Object {
                    "height": "auto",
                    "width": "auto",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#2B342C",
                        "fontFamily": "SofiaProRegular",
                        "fontSize": 12,
                        "height": 16,
                        "marginTop": 16,
                      },
                    ]
                  }
                >
                  Document(Proof of test)*
                </Text>
                <View>
                  <View
                    style={
                      Array [
                        Object {
                          "backgroundColor": "#FFFFFF",
                          "borderRadius": 4,
                          "flexDirection": "row",
                          "height": 48,
                        },
                        null,
                        Object {
                          "borderStyle": "dotted",
                          "borderWidth": 1,
                          "justifyContent": "center",
                          "marginTop": 8,
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "borderRadius": 4,
                            "height": 52,
                            "justifyContent": "center",
                            "shadowColor": "#DDDDDD",
                            "shadowOffset": Object {
                              "height": 8,
                              "width": 0,
                            },
                            "shadowOpacity": 0.75,
                            "shadowRadius": 4,
                          },
                          Object {
                            "height": "100%",
                            "shadowColor": "rgba(0,0,0,0)",
                            "width": "100%",
                          },
                        ]
                      }
                    >
                      <View
                        accessibilityLabel="ImagePicker"
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "backgroundColor": undefined,
                            "borderRadius": 4,
                            "flexDirection": "row",
                            "height": "100%",
                            "justifyContent": "center",
                            "opacity": 1,
                            "width": "auto",
                          }
                        }
                        testID="ImagePicker"
                      >
                        <Text
                          numberOfLines={1}
                          style={
                            Array [
                              Object {
                                "fontFamily": "SofiaProRegular",
                                "fontSize": 17,
                                "textAlign": "center",
                              },
                              Object {
                                "alignSelf": "center",
                                "color": "#2B342C",
                                "fontSize": 18,
                                "maxWidth": "75%",
                                "textAlign": "center",
                              },
                            ]
                          }
                        >
                          Upload photo
                        </Text>
                        <View
                          style={
                            Array [
                              Object {
                                "overflow": "hidden",
                              },
                              Array [
                                Object {
                                  "height": 18,
                                  "position": "absolute",
                                  "right": 18,
                                  "width": 18,
                                },
                                Object {
                                  "marginLeft": 32,
                                  "position": "relative",
                                },
                              ],
                            ]
                          }
                        >
                          <FastImageView
                            resizeMode="contain"
                            source={
                              Object {
                                "cache": "immutable",
                                "uri": undefined,
                              }
                            }
                            style={
                              Object {
                                "bottom": 0,
                                "left": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
  <View
    style={
      Object {
        "bottom": 158.74599999999998,
        "height": 100,
        "justifyContent": "space-between",
        "position": "absolute",
        "width": 722,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 4,
            "height": 52,
            "justifyContent": "center",
            "shadowColor": "#DDDDDD",
            "shadowOffset": Object {
              "height": 8,
              "width": 0,
            },
            "shadowOpacity": 0.75,
            "shadowRadius": 4,
          },
          Object {
            "backgroundColor": "#2B342C",
          },
        ]
      }
    >
      <View
        accessibilityLabel="Continue"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2B342C",
            "borderRadius": 4,
            "height": "100%",
            "justifyContent": "center",
            "opacity": 1,
            "width": 718,
          }
        }
        testID="Continue"
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontFamily": "SofiaProRegular",
                "fontSize": 17,
                "textAlign": "center",
              },
              Object {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          [missing "en.healthTest.continue" translation]
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 4,
            "height": 52,
            "justifyContent": "center",
            "shadowColor": "#DDDDDD",
            "shadowOffset": Object {
              "height": 8,
              "width": 0,
            },
            "shadowOpacity": 0.75,
            "shadowRadius": 4,
          },
          Object {
            "backgroundColor": "#FFFFFF",
            "marginTop": 16,
          },
        ]
      }
    >
      <View
        accessibilityLabel="skipTest"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 4,
            "height": "100%",
            "justifyContent": "center",
            "opacity": 1,
            "width": 718,
          }
        }
        testID="skipTest"
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontFamily": "SofiaProRegular",
                "fontSize": 17,
                "textAlign": "center",
              },
              Object {
                "color": "#2B342C",
              },
            ]
          }
        >
          [missing "en.healthTest.iDoNotHaveAnyTest" translation]
        </Text>
      </View>
    </View>
  </View>
  <Modal
    hardwareAccelerated={false}
    visible={false}
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#E5E5E5",
            "height": "100%",
            "width": "100%",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#F3F3F3",
            "height": "100%",
            "width": 750,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "top": 200.1,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "ReactComponent": [Function],
            }
          }
          style={
            Object {
              "marginBottom": "8%",
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#565655",
              "fontFamily": "Futura",
              "fontSize": 24,
              "fontWeight": "500",
              "marginBottom": "5%",
              "textAlign": "center",
            }
          }
        >
          [missing "en.STRINGS.ERROR_READING_QR_OOPS" translation]
        </Text>
        <Text
          style={
            Object {
              "marginHorizontal": "10%",
              "textAlign": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#8D8C91",
                "fontFamily": "arial",
                "fontSize": 16,
                "fontWeight": "400",
                "textAlign": "center",
              }
            }
          >
            [missing "en.STRINGS.ERROR_READING_QR_BEGIN" translation]
          </Text>
          <Text
            style={
              Object {
                "color": "#02BF8E",
                "fontFamily": "arial",
                "fontSize": 16,
                "fontWeight": "700",
                "textAlign": "center",
              }
            }
          >
            [missing "en.STRINGS.ME" translation]
          </Text>
          <Text
            style={
              Object {
                "color": "#8D8C91",
                "fontFamily": "arial",
                "fontSize": 16,
                "fontWeight": "400",
                "textAlign": "center",
              }
            }
          >
            [missing "en.STRINGS.ERROR_READING_QR_END" translation]
          </Text>
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "#007AFF",
                "fontFamily": "arial",
                "fontSize": 16,
                "fontWeight": "400",
                "textAlign": "center",
                "textDecorationLine": "underline",
              }
            }
          >
            [missing "en.STRINGS.CONTACT_SUPPORT" translation]
          </Text>
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "flex-end",
          }
        }
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#02BF8E",
              "borderRadius": 4,
              "bottom": 120.06,
              "elevation": 3,
              "height": 112.5,
              "justifyContent": "center",
              "opacity": 1,
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.22,
              "shadowRadius": 2.22,
              "width": 675,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 16,
                "fontWeight": "500",
                "textAlign": "center",
              }
            }
          >
            [missing "en.STRINGS.ERROR_TRY_AGAIN" translation]
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderColor": "#02BF8E",
              "borderRadius": 4,
              "borderWidth": 1,
              "bottom": 93.38000000000001,
              "height": 112.5,
              "justifyContent": "center",
              "opacity": 1,
              "width": 675,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#02BF8E",
                "fontSize": 16,
                "fontWeight": "500",
                "textAlign": "center",
              }
            }
          >
            [missing "en.STRINGS.CANCEL" translation]
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
